File: YFDebugPannel/YFDebugPannel/HCStepperCell.m
#import "HCStepperCell.h"
#import "HCCellItem.h"

@interface HCStepperCell ()

@property (nonatomic, strong) UIStepper *stepper;
@property (nonatomic, strong) UILabel *valueLabel;

@end

@implementation HCStepperCell

- (instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier {
    if (self = [super initWithStyle:UITableViewCellStyleSubtitle reuseIdentifier:reuseIdentifier]) {
        _stepper = [[UIStepper alloc] init];
        [_stepper addTarget:self action:@selector(stepperChanged:) forControlEvents:UIControlEventValueChanged];
        _valueLabel = [[UILabel alloc] initWithFrame:CGRectMake(0, 0, 40, 24)];
        _valueLabel.textAlignment = NSTextAlignmentRight;
        UIStackView *stack = [[UIStackView alloc] initWithArrangedSubviews:@[_valueLabel, _stepper]];
        stack.axis = UILayoutConstraintAxisHorizontal;
        stack.spacing = 8.0;
        self.accessoryView = stack;
        self.selectionStyle = UITableViewCellSelectionStyleNone;
    }
    return self;
}

- (void)configureWithItem:(HCCellItem *)item minimum:(NSInteger)min maximum:(NSInteger)max {
    self.textLabel.text = item.title;
    self.detailTextLabel.text = item.desc;
    self.stepper.minimumValue = min;
    self.stepper.maximumValue = max;
    self.stepper.enabled = item.enabled;
    NSInteger value = [item.value integerValue];
    self.stepper.value = value;
    self.valueLabel.text = [NSString stringWithFormat:@"%ld", (long)value];
}

- (void)stepperChanged:(UIStepper *)sender {
    NSInteger value = (NSInteger)sender.value;
    self.valueLabel.text = [NSString stringWithFormat:@"%ld", (long)value];
    if (self.valueChanged) {
        self.valueChanged(value);
    }
}

@end

File: YFDebugPannel/YFDebugPannel/HCStepperCell.swift
import UIKit

final class HCStepperCell: UITableViewCell {
    var valueChanged: ((Int) -> Void)?

    private let stepper = UIStepper()
    private let valueLabel = UILabel()
    private let stackView = UIStackView()

    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
        super.init(style: .subtitle, reuseIdentifier: reuseIdentifier)
        stepper.addTarget(self, action: #selector(stepperChanged(_:)), for: .valueChanged)
        valueLabel.textAlignment = .right
        valueLabel.setContentHuggingPriority(.required, for: .horizontal)
        valueLabel.setContentCompressionResistancePriority(.required, for: .horizontal)
        stackView.axis = .horizontal
        stackView.spacing = 8
        stackView.alignment = .center
        stackView.addArrangedSubview(valueLabel)
        stackView.addArrangedSubview(stepper)
        valueLabel.widthAnchor.constraint(equalToConstant: 40).isActive = true
        accessoryView = stackView
        selectionStyle = .none
    }

    required init?(coder: NSCoder) {
        super.init(coder: coder)
    }

    func configure(with item: HCCellItem, minimum: Int, maximum: Int) {
        textLabel?.text = item.title
        detailTextLabel?.text = item.desc
        stepper.minimumValue = Double(minimum)
        stepper.maximumValue = Double(maximum)
        stepper.isEnabled = item.enabled
        let value = intValue(item.value)
        stepper.value = Double(value)
        valueLabel.text = String(value)
    }

    @objc private func stepperChanged(_ sender: UIStepper) {
        let value = Int(sender.value)
        valueLabel.text = String(value)
        valueChanged?(value)
    }
}
